subprojects/persistent-cache/src/main/java/org/gradle/cache/internal/LeastRecentlyUsedCacheCleanup.java

```diff
@@ -33,26 +34,29 @@ public class LeastRecentlyUsedCacheCleanup extends AbstractCacheCleanup {
     private static final Logger LOGGER = LoggerFactory.getLogger(LeastRecentlyUsedCacheCleanup.class);
 
     public static final long DEFAULT_MAX_AGE_IN_DAYS_FOR_RECREATABLE_CACHE_ENTRIES = 7;
-    public static final long DEFAULT_MAX_AGE_IN_DAYS_FOR_EXTERNAL_CACHE_ENTRIES = 30;
 
     private final FileAccessTimeJournal journal;
-    private final long minimumTimestamp;
+    private final Supplier<Integer> numberOfDays;
```

Comment:
Can this be a `Provider` instead, to remove the need for creating a lambda to pass which just calls `.get()` on the `Property`?

I kind of a think there's no great reason to change this from a long (or int).  I understand we want everything to be lazy going forward, but it's hard to imagine a use case for lazily setting the number of days something remains in these caches, and doing it from an initscript.

Reply:
This is in persistent-cache which has no knowledge of `Provider` and is further up the dependency chain.  My original plan was to just pass the provider through here, but since it's not visible in this module (and shouldn't be) we just use the supplier instead.

The reason this is a provider is that there are places where we need to supply a reference to this value in a few places before it can be set.  We _could_  handle this laziness through a supplier that directly references some value in the cache configuration, but we also had the need to finalize the values, which is already implemented in `Provider`.